### PR TITLE
Fetch prefix one time for a batch

### PR DESCRIPTION
**Pull request by hweawer**

### CODE REVIEW

File: src/bots/unvetter.py [hunk 1]
```diff
@@ -146,5 +147,8 @@ def _send_unvet_message(self, message: UnvetMessage) -> bool:
     def _clear_outdated_messages_for_module(self, module_id: int, nonce: int) -> None:
         if self.message_storage is None:
             return
-
-        self.message_storage.get_messages(lambda message: message['stakingModuleId'] != module_id or message['nonce'] >= nonce)
+        prefix = self.w3.lido.deposit_security_module.get_unvet_message_prefix()
+        sign_filter = get_messages_sign_filter(prefix)
+        self.message_storage.get_messages_and_actualize(
+            sign_filter, lambda message: message['stakingModuleId'] != module_id or message['nonce'] >= nonce
```
Comment (F4ever) [inline src/bots/unvetter.py:153]: Maybe just declare filter method here?
```
prefix = self.w3.lido.deposit_security_module.get_unvet_message_prefix()
is_prefix_valid = get_messages_sign_filter(prefix)

def unvet_filter(message: Message) -> bool:
    is_message_relevant = (
        message['stakingModuleId'] != module_id or 
        message['nonce'] >= nonce
     )
     return is_message_relevant and is_prefix_valid(msg)

self.message_storage.get_messages_and_actualize(unvet_filter)
```

File: src/bots/unvetter.py [hunk 2]
```diff
@@ -146,5 +147,8 @@ def _send_unvet_message(self, message: UnvetMessage) -> bool:
     def _clear_outdated_messages_for_module(self, module_id: int, nonce: int) -> None:
         if self.message_storage is None:
```
Comment (F4ever) [inline src/bots/unvetter.py:148]: Seems we can remove this condition

File: src/bots/unvetter.py [hunk 3]
```diff
@@ -90,12 +89,14 @@ def execute(self, block: BlockData) -> bool:
 
         return True
 
-    def receive_unvet_messages(self) -> list[UnvetMessage]:
+    def receive_unvet_messages(self) -> Iterable[UnvetMessage]:
```
Comment (F4ever) [inline src/bots/unvetter.py:92]: In the return annotation, it is best to describe the type as precisely as possible.
Knowing that there is not just iterable, but list we can, for example, just call len method without any casting.
In the input annotations better to use minimal requirements for values. For example, if we're just iterating over an object, there is no need to accept list, we can accept any iterable object

File: src/transport/msg_storage.py [hunk 4]
```diff
@@ -1,40 +1,55 @@
...
+                                      used to filter messages when they are received.
         """
         self._transports = transports
         self._filters = filters
 
-    def receive_messages(self) -> Iterable[dict]:
-        """Fetch all messages from transport and filter them"""
-        for transport in self._transports:
-            messages = transport.get_messages()
+    def get_messages_and_actualize(self, *args: Callable[[BotMessage], bool]) -> Iterable[BotMessage]:
```
Comment (F4ever) [inline src/transport/msg_storage.py:30]: Better, not use *args. Let's just make it Iterable[Callable].
Or let's just receive one functions here (but with required filters inside) as in example in prev comment

**Comment by F4ever** [comment]
Fix pyright checks pls

File: src/transport/msg_types/common.py [hunk 5]
```diff
@@ -79,12 +78,11 @@ def _verification_data_deposit(prefix: bytes, msg: DepositMessage) -> tuple[List
 
 
 def _verification_data_pause(prefix: bytes, msg: PauseMessage) -> tuple[List[Any], List[str]]:
-    if msg.get('stakingModuleId', -1) != -1:
-        data = [prefix, msg['blockNumber'], msg['stakingModuleId']]
-        abi = ['bytes32', 'uint256', 'uint256']
-    else:
-        data = [prefix, msg['blockNumber']]
-        abi = ['bytes32', 'uint256']
+    data = [prefix, msg['blockNumber']]
```
[the remark this answers is not in the document]
Comment (hweawer) [reply src/transport/msg_types/common.py]: Reverted

File: src/bots/unvetter.py (hunk 3 above)
Comment (hweawer) [reply src/bots/unvetter.py]: Reverted

File: src/transport/msg_storage.py (hunk 4 above)
Comment (hweawer) [reply src/transport/msg_storage.py]: Replaced

File: src/bots/unvetter.py [hunk 6]
```diff
@@ -143,8 +144,12 @@ def _send_unvet_message(self, message: UnvetMessage) -> bool:
...
         return result
 
-    def _clear_outdated_messages_for_module(self, module_id: int, nonce: int) -> None:
-        if self.message_storage is None:
-            return
+    def _clear_outdated_messages_for_module(self, module_id: int, nonce: int):
+        prefix = self.w3.lido.deposit_security_module.get_unvet_message_prefix()
+        sign_filter = get_messages_sign_filter(prefix)
+
+        def unvet_filter(msg: TypedDict) -> bool:
```
Comment (F4ever) [inline src/bots/unvetter.py:151]: is_unvet_message_actual

File: src/bots/unvetter.py [hunk 7]
```diff
@@ -143,8 +144,12 @@ def _send_unvet_message(self, message: UnvetMessage) -> bool:
...
+            is_message_relevant = msg['stakingModuleId'] != module_id or int(msg['nonce']) >= nonce
+            return is_message_relevant and sign_filter(msg)
```
Comment (F4ever) [inline src/bots/unvetter.py:153]: add some verb to sign_filter

File: src/bots/unvetter.py (hunk 6 above)
Comment (hweawer) [reply src/bots/unvetter.py]: Done

**Review by F4ever — APPROVED** [review]

**Pull request merged**